pkg/resources/management.cattle.io/v3/projectroletemplatebinding/validator.go

```diff
@@ -94,6 +94,11 @@ func (a *admitter) Admit(request *admission.Request) (*admissionv1.AdmissionResp
 		return nil, fmt.Errorf("failed to decode PRTB object from request: %w", err)
 	}
 
+	clusterNS, projectNS := clusterFromProject(prtb.ProjectName)
+	if projectNS != prtb.Namespace {
```

Comment:
I think this might be at the wrong spot in the validation process. I think in the current format, if you try to change the `projectName` on an active PRTB, chances are that you will hit this validation (saying that the `projectName` doesn't match the namespace), rather than the "right" validation (that you can't change projectName after creation). 

I'd recommend moving this to the `validateCreateFields` function rather than having it in the main flow here.

Reply:
Good point. I mostly wanted to avoid calling `clusterFromProject` twice, but it's cheap, so that's fine. Will adjust.